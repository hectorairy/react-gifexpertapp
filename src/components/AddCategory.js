import React, { useState } from "react";
import PropTypes from "prop-types";

export default function AddCategory({ setCategories }) {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // * No es necesario mandar las categorias, puede hacerlo unicamente con la funciÃ³n
    // * setCategories([...categories, inputValue]);
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
