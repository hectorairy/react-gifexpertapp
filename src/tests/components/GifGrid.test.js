import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente <GifGrid />", () => {
  const category = "One punch";
  test("Debe de hacer match en el componente <GifGrid />", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/algo.jpg",
        title: "Algo",
      },
      {
        id: "ABCD",
        url: "https://localhost/algo.jpg",
        title: "Algo",
      },
      {
        id: "ABCE",
        url: "https://localhost/algo.jpg",
        title: "Algo",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
