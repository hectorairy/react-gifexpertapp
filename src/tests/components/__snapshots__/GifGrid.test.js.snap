// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de hacer match en el componente <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
     
    One punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
     
    One punch
     
  </h3>
  <div
    className="cardGrid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="https://localhost/algo.jpg"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="Algo"
      url="https://localhost/algo.jpg"
    />
    <GifGridItem
      id="ABCE"
      key="ABCE"
      title="Algo"
      url="https://localhost/algo.jpg"
    />
  </div>
</Fragment>
`;
